kernel/kernel-api/src/test/scala/io/delta/kernel/internal/tablefeatures/TableFeaturesSuite.scala

```diff
@@ -944,6 +944,63 @@ class TableFeaturesSuite extends AnyFunSuite {
       }
   }
 
+  test(
+    "updateProtocolWithFeaturOptionsOverrides add missing features and removes them from metadta") {
+    val protocol = new Protocol(3, 7).withFeature(TableFeatures.APPEND_ONLY_W_FEATURE)
+    val metadata = testMetadata(tblProps = Map(
+      "delta.feature.deletionVectors" -> "supported",
+      "delta.feature.appendOnly" -> "supported",
```

Comment:
Add a feature (rowTracking or icebergCompatV2) that has dependent features and ensure they are added to the output.

Reply:
Unknown feature throws error.